example/lib/network_events.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+import 'package:fullstory_flutter/dio.dart';
+
+/// Demo widget that sends network requests using package:dio and package:http.
+class NetworkEvents extends StatelessWidget {
+  const NetworkEvents({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        TextButton(
+          onPressed: () => _makeDioRequest(context),
+          child: const Text('Send dio request'),
+        ),
+      ],
+    );
+  }
+
+  Future<void> _makeDioRequest(BuildContext context) async {
+    final response = await dio.get('https://fullstory.com');
+    if (context.mounted) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          content: Text('Dio got response: ${response.statusCode}'),
+        ),
+      );
+    }
+  }
+}
+
+// This is declared at the top-level to allow [NetworkEvents] to be const.
+Dio get dio => Dio()..interceptors.add(FSInterceptor());
```

Comment:
This seems like the most important part - could we move it to the top of the file, or does it need to be here at the end?